examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt

```diff
@@ -207,13 +207,16 @@ class BarcodeFragment : Fragment() {
             lateinit var payload: SetupPayload
             try {
                 payload = SetupPayloadParser().parseQrCode(barcode.displayValue)
+                FragmentUtil.getHost(this@BarcodeFragment, Callback::class.java)
```

Comment:
so this says "get Callback from BarcodeFragment" and the exception you get is "IllegalState: BarcodeFragment is not a callback" ... so why do we insist on using this then?